extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSourceBuilder.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic;
+
+import com.hazelcast.function.ConsumerEx;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.core.Vertex;
+import com.hazelcast.jet.elastic.impl.ElasticProcessorMetaSupplier;
+import com.hazelcast.jet.pipeline.BatchSource;
+import com.hazelcast.jet.pipeline.Sources;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHit;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+
+import static com.hazelcast.jet.impl.util.Util.checkSerializable;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Builder for Elasticsearch source which reads data from Elasticsearch and
+ * converts SearchHits using provided {@code mapHitFn}
+ *
+ * @param <T> type of the mapping function from {@link SearchHit} -> T
+ *           TODO not sure about the type parameter name - T as the usual default, or R for Result
+ *           also we could accept the function in the build() method, same as the original source did it
+ * @since 4.1
+ */
+public class ElasticSourceBuilder<T> implements Serializable {
```

Comment:
Builders shouldn't be serializable. We don't want it capturing anything extra by accident. It makes the seperation between what's part of source params and what's not (i.e. parallelism) unclear.

Reply:
Moved the configuration to Serializable config class.